### ISSUE TITLE
[Backend] Replace explicit DTOs with Projection Interfaces

### ISSUE THREAD

**Issue by dnl50** [adessoAG/project-board#30]
Spring has the ability to map a **given** entity by defining projection interfaces similar to Spring Data REST projections by using a [`ProjectionFactory`](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/projection/ProjectionFactory.html) implementation.

The [`SpelAwareProxyProjectionFactory`](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/projection/SpelAwareProxyProjectionFactory.html) even allows using SpEL expressions on the interface methods.

# Example

A projection interface for a `UserData` entity could look like this:

```java

public interface UserIdAndFirstNameProjection {

    @Value("#{target.user.id})
    String getUserId();

    String getFirstName();

}

```

When adapting the `UserController` to use a `SpelAwareProxyProjectionFactory` to map these entities, the resulting projection after serializing it to JSON would look like this:

```json

{
    "userId": "jdoe",
    "firstName": "Jane"
}

```

The adapted `getUserById(String userId, ...)` could look like this:

```java

@GetMapping(path = "/{userId}")
public ResponseEntity<?> getUserById(@PathVariable String userId) {
    var projectionFactory = new SpelAwareProxyProjectionFactory();
    var user = userService.getUserById(userId);
    var userData = userService.getUserData(user);

    var respContent = projectionFactory.createProjection(UserIdAndFirstNameProjection.class, userData);

    return ResponseEntity.ok(respContent);
}

```

**Comment by thetric** [comment]
This could be simplified by directly using projections in the Spring Data repositories ([Spring Data JPA reference](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#projections)).

**Comment by dnl50** [comment]
I decided to implement it this way because I didn't want to use projections in the service layer.

**Comment by dnl50** [comment]
Implemented in 5eac606
